eng/scripts/helpers.js

```diff
@@ -21,6 +21,9 @@ function run(command, args, options) {
 
   if (process.platform === "win32" && isCmdOnWindows.includes(command)) {
     command += ".cmd";
+    // When spawning .bat or .cmd files on windows, node will now error with EINVAL if shell is not set to true.
+    // https://nodejs.org/en/blog/vulnerability/april-2024-security-releases-2
+    options.shell = true;
```

Comment:
hmm... how deep do you think the change will go then?

Reply:
there is 3 options:
1. One this isn't used anywhere where it is a problem so we can keep it
2. Anywhere it is a problem we escape manually
3. We use `cross-spawn` pacakge instead but autorest still using rush it is very anoying to have top level dev dependency for scripts.

Reply:
was there something in particular that broke here?

Reply:
Should we push the escaping requirement to the caller and error here if they haven't already set shell=true?

Reply:
> was there something in particular that broke here?

Autorest regeneration fails consistently on a single project in azure-sdk-for-net:

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3806494&view=logs&j=22d114d9-e1b7-5381-70ab-ddaabfa32a9d&t=1eb743b7-6600-5425-47d2-d44920e345d6&l=2235
```
  fatal   | Process() cancelled due to failure 
EXEC : error   | error : spawn EINVAL [D:\a\_work\1\s\azure-sdk\sdk\resourcehealth\Azure.ResourceManager.ResourceHealth\src\Azure.ResourceManager.ResourceHealth.csproj]
  error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
```

Reply:
> Should we push the escaping requirement to the caller and error here if they haven't already set shell=true?

yeah we could do that,


so I doubt this will fix the error above. This file is only used for intenral scripts